src/decoder.rs

```diff
@@ -120,6 +123,39 @@ pub fn strip_junk_header(slice: &[u8]) -> io::Result<&[u8]> {
     Ok(&slice[slice.len()..])
 }
 
+/// Decodes range mappping bitfield string into index
+fn decode_rmi(rmi_str: &str, val: &mut BitVec<u8, Msb0>) -> Result<()> {
+    val.clear();
+    if val.capacity() < rmi_str.len() * 6 {
+        val.reserve(rmi_str.len() * 6 - val.capacity());
+    }
+    unsafe {
+        // Safety: we just reserved enough space
+        val.set_len(rmi_str.len() * 6);
+    };
+
+    // A: 0b000000
+    // B: 0b000001
+
+    // While,
+    // AAB: 0b000000 0b000000 0b000001 => 13
+    // so we iterate in the reverse order
+    for (idx, &byte) in rmi_str.as_bytes().iter().rev().enumerate() {
+        let byte = match byte {
+            b'A'..=b'Z' => byte - b'A',
+            b'a'..=b'z' => byte - b'a' + 26,
+            b'0'..=b'9' => byte - b'0' + 52,
+            b'+' => 62,
+            b'/' => 63,
+            _ => unreachable!("invalid rmi"),
+        };
+
+        val[6 * idx..6 * (idx + 1)].store::<u8>(byte);
+    }
+
+    Ok(())
+}
```

Comment:
Much better! However, I think this is simpler and clearer if you use `Lsb0` bit order and `store_le/load_le`. For example, take the input string `"DAB"`. `D` is 3, or `000011`, A is `000000`, and `B` is `000001`. In total, this means that mappings 0, 1, and 12 (counting from 0) are range mappings.

Decoding this with the current implementation gives the bit vector `[0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 0]`, i.e. bits 5, 14, and 15 are set. It's really hard to understand the correspondence between this bit vector and the original encoding.

By contrast, if you use `Lsb0` and skip the `.rev()` in the for loop, you get the vector `[1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0]`, which is exactly what you would expect (bits 0, 1, 12 set).

Reply:
Thank you! It worked!